buildflow/core/app/runtime/actors/runtime.py

```diff
@@ -141,13 +163,30 @@ def is_active(self):
     async def _runtime_autoscale_loop(self):
         logging.info("Runtime checkin loop started...")
         while self._status == RuntimeStatus.RUNNING:
-            for processor_pool in self._processor_pool_actors:
+            for processor_pool in self._processor_pool_refs:
                 if self._status != RuntimeStatus.RUNNING:
                     break
 
-                processor_snapshot: ProcessorSnapshot = (
-                    await processor_pool.snapshot.remote()
-                )
+                try:
+                    processor_snapshot: ProcessorSnapshot = (
+                        await processor_pool.actor_handle.snapshot.remote()
+                    )
+                except RayActorError:
+                    logging.exception("process actor unexpectedly died. will restart.")
+                    processor_pool.actor_handle = (
+                        PipelineProcessorReplicaPoolActor.remote(
+                            self.run_id,
+                            processor_pool.processor,
+                            processor_pool.options,
```

Comment:
nit: Consider looking up the processor options from the runtime options (keyed by processor id, and get lookup with processor_pool.proccessor.processor_id)

Reply:
done.